webhook/notifier.go

```diff
@@ -71,3 +73,151 @@ func (n *DefaultNotifier) RegisterProcessedHook(hook func(ctx context.Context, w
 		u.RegisterProcessedHook(hook)
 	}
 }
+
+// ---------------------------------
```

Comment:
consolidating common functions/structures between different notifiers into this file.